soccer/optimization/NelderMead2DTest.cpp

```diff
@@ -0,0 +1,56 @@
+#include <gtest/gtest.h>
+#include "NelderMead2D.hpp"
+#include "NelderMead2DConfig.hpp"
+#include <Geometry2d/Util.hpp>
+#include <iostream>
+
+static float evalFunction1(Geometry2d::Point p) {
+    return -1 * sqrt(p.x() * p.x() + p.y() * p.y());
+}
+
+static float evalFunction2(Geometry2d::Point p) { return 1; }
+
+TEST(NelderMead2D, execute) {
+    std::function<float(Geometry2d::Point)> f = &evalFunction1;
+    NelderMead2DConfig config(
+        &f, Geometry2d::Point(1, 1), Geometry2d::Point(1, 1),
+        Geometry2d::Point(0.001, 0.001), 1, 2, .5, .5, 100, 0, 0);
+
+    NelderMead2D nm(&config);
+
+    nm.execute();
+
+    EXPECT_NEAR(nm.getValue(), 0, 0.001);
+    EXPECT_NEAR(nm.getPoint().x(), 0, 0.001);
+    EXPECT_NEAR(nm.getPoint().y(), 0, 0.001);
+}
+
+TEST(NelderMead2D, iteration_limit) {
+    std::function<float(Geometry2d::Point)> f = &evalFunction2;
+    NelderMead2DConfig config(
+        &f, Geometry2d::Point(1, 1), Geometry2d::Point(.0001, .0001),
```

Comment:
I don't think so. Since we need the address of the std::function, we would need the address of the address of the function pointer `evalFunction2`. This was the cleanest way I could think of.

Do you remember that video that you showed everyone at the beginning of the year? It was like 4-5 minutes long and how a function pointer could be deferenced multiple times without changing the meaning to make the code look super complicated, eg: `****evalFunction2`. Does std::function work the same way?